cmd/s3-peer-client.go

```diff
@@ -19,191 +19,185 @@ package cmd
 import (
 	"encoding/json"
 	"fmt"
-	"net/rpc"
 	"net/url"
 	"path"
 	"sync"
-	"time"
 )
 
+// s3Peers structs contains the peer addresses in the cluster, and
+// their corresponding BucketMetaState interface objects.
 type s3Peers struct {
-	// A map of peer server address (in `host:port` format) to RPC
-	// client connections.
-	rpcClients map[string]*AuthRPCClient
-
-	mutex *sync.RWMutex
+	// Collection of peers
+	peers struct {
+		// address of peer in `host:port` format
+		addr []string
+		// BucketMetaState object for peers
+		bmsClient []BucketMetaState
+	}
```

Comment:
this is a case for:

`struct s3Peer {addr string, bmsClient BucketMetaState}`

`peers []struct s3Peer`

makes the code more readable at the cost of an extra type (s3Peer)